packages/cli/src/commands/auth/login.ts

```diff
@@ -116,4 +128,29 @@ export default class Login extends Command {
   private get configuration() {
     return new Config(this.config.configDir, this.error);
   }
+
+  private async getAllOrgsUserHasAccessTo() {
+    const orgs = await (
+      await new AuthenticatedClient().getClient()
+    ).organization.list();
+    return (orgs as unknown) as OrganizationModel[];
+  }
+
+  private async verifyOrganization() {
+    const flags = this.flags;
+    const orgs = await this.getAllOrgsUserHasAccessTo();
+
+    if (flags.organization) {
+      const found = orgs.find((o) => o.id === flags.organization);
+      if (!found) {
+        this.error(
+          `You either do not have access to organization ${flags.organization}, or it does not exists.`
+        );
+      }
+    }
+
+    if (orgs.length === 0) {
+      this.error('You do not have access to any Coveo organization.');
```

Comment:
Maybe the user chose the wrong region or environment. So maybe we can say something like "no orgs available in this region/env"?